src/CarbonAware.WebApi/test/integrationTests/CarbonAwareControllerTests.cs

```diff
@@ -0,0 +1,99 @@
+namespace CarbonAware.WepApi.IntegrationTests;
+
+using CarbonAware.Tools.WattTimeClient;
+using Microsoft.AspNetCore.Mvc.Testing;
+using NUnit.Framework;
+using System.Net;
+using WireMock.Server;
+
+public class APIWebApplicationFactory : WebApplicationFactory<Program> {
+
+}
+
+/// <summary>
+/// Tests that the Web API controller handles and packages various responses from a plugin properly 
+/// including empty responses and exceptions.
+/// </summary>
+[TestFixture]
+public class CarbonAwareControllerTests
+{
+#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+	private APIWebApplicationFactory _factory;
+	private HttpClient _client;
+    private WireMockServer _server;
+#pragma warning restore CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+    private bool resetEnvironment = false;
+
+    [OneTimeSetUp]
+    public void Setup()
+    {
+		//Defaults to JSON Integration Testing
+		//WattTime can be tested by uncommenting the line below or setting an env variable
+		//Environment.SetEnvironmentVariable("CarbonAwareVars__CarbonIntensityDataSource", "WattTime");
+
+		if (Environment.GetEnvironmentVariable("CarbonAwareVars__CarbonIntensityDataSource") == "WattTime")
+        {
+            //Start WireMock server and point the integration tests to it for mocking outbound calls
+            //Only for WattTime Integration Testing
+            _server = WireMockServer.Start();
+            _server.SetupWattTimeServerMocks();
```

Comment:
General design feedback:  I think we should encourage the tests to define their mock WattTime responses rather than hard coding static values for every test.

Extension methods (similar to what you have here) can be a great way to abstract the boilerplate and create a clean interface to tersely define the mock under test.

EG
```
var responseData = new EmissionsForecast();
_server.MockDataEndpoint(HttpStatusCode.Successful, responseData);
```

Reply:
Just pushed up the ability to do this, tbd how it integrates into the broader discussion we had this morning